packages/block-library/src/navigation-link/edit.js

```diff
@@ -183,7 +182,7 @@ function NavigationLinkEdit( {
 				</div>
 				<InnerBlocks
 					allowedBlocks={ [ 'core/navigation-link' ] }
-					renderAppender={ hasDescendants ? InnerBlocks.ButtonBlockAppender : false }
+					renderAppender={ isSelected ? InnerBlocks.ButtonBlockAppender : false }
```

Comment:
Shouldn't this have the two conditions?

Reply:
Why? We would want the appender to show even if there are no descendants right?

Reply:
Not necessarily. Top level items don't show an appender until you click on the "create sub-menu" option. I'd expect it to be similar for sub-menus.

Reply:
This seems okay to me since the user has found that button once already, they'd know to repeat for more levels.

Reply:
I see. But then why show an appender at all? Can't we always rely on that one button?

Reply:
If an item already has sub-items, the button doesn't make much sense as there is not a trailing inserter like in every other place that has child blocks present.